go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/openshift/check-payload
 
-go 1.19
+go 1.21
```

Comment:
1. The commit title is somewhat deceiving -- this is more than `go mod tidy`, you are bumping the minimally supported Go version. Please say so.

2. If you're bumping it, you need to remove `internal/golang/elf_go120.go` as it is no longer needed. Maybe there's something more.

3. If you are to bump it, why not bump to Go 1.23 right away (Go 1.22 is no longer supported).

4. It makes sense to also modify CI (`.github/workflows/validate.yml`) to test on supported Go versions (1.23.x, 1.24.x).

5. I did the same change locally, but it did not result in the same change to imdario/mergo that your commit did. Hmm

Reply:
6. Go version in Dockerfile might need to be updated, too.

Reply:
My fault. Those are entirely valid points and this commit doesn't align with this PR--I'll move it to a separate PR for review.

The reproducer for this commit is:
- Remove the `replace` statement (aside from being cleaner, its removal should enable a `go install` of the binary)
- `go get github.com/imdario/mergo@v0.3.16`
- `go mod tidy`

I honestly wasn't expecting the Go bump with `go mod tidy`, but it's definitely inappropriate for this PR.

Reply:
Opened up a new PR for the Go bump, including items listed (and other instances of Go versions I found in devcontainer and `go:build` statement in the `elf` file):
- #252